src/components/PageDefinitions.js

```diff
@@ -181,62 +219,197 @@ class PageDefinitions extends Component {
     this.refs.contributeModal.open()
   }
 
-  facetChange(value) {
-    const activeFacets = (value || []).map(facet => facet.value)
-    this.setState({ ...this.state, activeFacets })
+  name = coordinates => {
+    return coordinates.name ? coordinates.name : null
+  }
+
+  namespace = coordinates => {
+    return coordinates.namespace ? coordinates.namespace : null
+  }
+
+  provider = coordinates => {
+    return coordinates.provider ? coordinates.provider : null
+  }
+
+  type = coordinates => {
+    return coordinates.type ? coordinates.type : null
+  }
+
+  releaseDate = coordinates => {
+    const definition = this.props.definitions.entries[EntitySpec.fromCoordinates(coordinates).toPath()]
+    const described = get(definition, 'described')
+    if (described && described.releaseDate) return described.releaseDate
+    return null
+  }
+
+  license = coordinates => {
+    const definition = this.props.definitions.entries[EntitySpec.fromCoordinates(coordinates).toPath()]
+    const licensed = get(definition, 'licensed')
+    if (licensed && licensed.declared) return licensed.declared
+    return null
+  }
+
+  getSort(eventKey) {
+    switch (eventKey.value) {
+      case 'name':
+        return this.name
+      case 'namespace':
+        return this.namespace
+      case 'provider':
+        return this.provider
+      case 'type':
+        return this.type
+      case 'releaseDate':
+        return this.releaseDate
+      case 'license':
+        return this.license
+    }
+  }
+
+  onSort(eventKey) {
+    this.setState({ ...this.state, activeSort: eventKey.value, counter: this.state.counter + 1 })
+    this.props.dispatch(uiBrowseUpdateList({ sort: this.getSort(eventKey) }))
+  }
+
+  filterList(list) {
+    const { activeFilters } = this.state
+    if (activeFilters.length === 0) {
+      return list
+    }
+    return filter(list, component => {
+      const defintion = this.getDefinition(component)
+      for (let filterType in activeFilters) {
+        const value = activeFilters[filterType]
+        const fieldValue = this.getValue(defintion, filterType)
+        if (value === 'presence') {
+          if (!fieldValue) return false
+        } else if (value === 'absence') {
+          if (fieldValue) return false
+        } else {
```

Comment:
`return fieldValue && fieldValue.toLowerCase().includes(value.toLowerCase())`

getting rid of negatives and nested else if etc makes the code easier to understand.